boa_engine/src/vm/opcode/mod.rs

```diff
@@ -121,16 +124,89 @@ where
     unsafe { read_unchecked(bytes, offset) }
 }
 
+/// Represents a varying operand kind.
+#[derive(Default, Debug, Clone, Copy)]
+#[allow(missing_docs)]
+pub enum VaryingOperandKind {
+    #[default]
+    Short,
+    Half,
+    Wide,
+}
```

Comment:
Is there a reason that these structs are `pub`?

I would be in favor of making `BindingOpcode`, `Instruction`, `InstructionIterator`, `Opcode` and `VaryingOperandKind` all `pub(crate)`. That way we can also remove the `missing_docs`.